index.html

```diff
@@ -519,6 +519,79 @@ <h3>LinkedDataSignature2016</h3>
           </pre>
       </section>
 
+
+      <section id="MerkleProof2019" about="https://w3id.org/security#MerkleProof2019">
+
+        <h3>MerkleProof2019</h3>
+          <p>
+A Linked Data signature is used for digital signatures on RDF Datasets.
+The default canonicalization mechanism is specified in the RDF Dataset
+Normalization specification, which effectively deterministically names all
+unnamed nodes. The default signature mechanism
+uses a SHA-256 digest and ECDSA to perform the digital signature. More information at
```

Comment:
> The default signature mechanism uses a SHA-256 digest and ECDSA to perform the digital signature.

Not sure if this is true; I don't think it is. The proofValue encodes the anchoring point on the blockchain, right? You don't use SHA-256 or ECDSA, right?

Reply:
so I based this description on this introduction here: https://w3c-ccg.github.io/lds-merkle-proof-2019/#introduction. 

The content (of the certificate, and furthermore of the merkle tree) is hashed before being saved on the blockchain. While I have worked less on the implementation of the issuance, during verification, to compare the local certificate to the one that was saved on the blockchain (via the proof), we hash with sha-256 to retrieve the expected value: https://github.com/blockchain-certificates/cert-verifier-js/blob/master/src/inspectors/computeLocalHash.js#L108.

However now that I am reading the code again, I think I made a mistake on the canonicalization algorithm: https://github.com/blockchain-certificates/cert-verifier-js/blob/master/src/inspectors/computeLocalHash.js#L84

Reply:
If you don't use ECDSA in the proof you should remove the term.... probably saying something like "The default proof mechanism relies on merkleproofs"

Reply:
I looked at the code, the merkle proof does not use any digital signatures, it does use sha-256.

Reply:
Hey @lemoustachiste @msporny @OR13 ,

I recall basing **MerkleProof2019** definition off the 2017 one.

ECDSA part refers to the process of verification of the credential since it is verified by storing a value (MerkleRoot hash) in a Blockchain transaction's input.

ECDSA is the algorithm that is used to sign the transaction using a wallet and I believe it can remain in the specification (potentially as a default one).

In this case in order to verify that the transaction with _transactionHash_ stored in the _proofValue_ was indeed signed with a wallet that belongs to a certain _DID_ you would use ECDSA to verify the blockchain transaction signature.

Does the explanation make sense?

Reply:
```
"proofValue": "zMcm4LfQFUZkWZyLpVrzYzaHRssD3xGQ6Yh6nEhErjfAqnxDWo7gBawdhynnbiucoqqwye54oTs5p6frH385nVvBf2UXMgRVtBZLq6tHN1MY1hQBMBBtXbWbYzj99GotAJRs7PxcvQmUQKHs8hkokZhgw92CNAtvDn2jeycvuyFPtWorxZaGP23mDkgwpberKmAqcPcsUFNooGe1UkQj8wws8D9pdpAVmuKf5Zo78iq1tUWSUdGwN9sK57SQP6YTtL59b7WUK1NEHD1GL8o5Z2mY6MnxuxfiqkpjepaPVqTPfgmaCuS",
    "proofPurpose": "assertionMethod",
    "verificationMethod": "ecdsa-koblitz-pubkey:0x7e30a37763e6Ba1fFeDE1750bBeFB4c60b17a1B3"
```

how do I get the transaction from this so I can do ECRecover / ECDSA verify to confirm the credential is anchored?

I think the part I am getting tripped up on is that a "merkle proof" does not use anything but sha-256.... and verifying a blockchain transaction that anchored a merkle root is not necessary to verify a merkle proof.

I'm not a merkle proof expert, so maybe I am wrong about this...

aren't you hashing the credential to a digest, batching the credential digest together with many others, and issueing a receipt, so you can prove that:

credential digest + receipt => is anchored in merkle root.

the merkle root could be anywhere, and anyone who has the merkle root and the receipt can verify the credential.

I assume the proofValue is the receipt, but then how do I get to the merkle root?

Reply:
I would expect `verificationMethod` to point to a blockchain transaction that anchored a merkle root, not a public key or ethereum address that could have been used to sign a transaction...

Reply:
@OR13 this is a very good question.
I will try to make a more detailed job explaining how both are connected tomorrow.

In a nutshell a Merkle Tree is an algorithm by which you can create a set of hashes that will lead down to the *leaf* which is a SHA256 of a particular credential in an array of credentials (to allow batch issuance of the credentials and confirming many credentials with one blockchain transaction).

The *merkleRoot* of the MerkleTree is another SHA256 that is not necessarily equal to the *targetHash*.

Now, these values partially create *proofValue* which is a MultiBase hash of a CBOR encoded object.

The other part is the transactionHash of the blockchain transaction.

After adding the transactionHash to the proofValue object it gets encoded.

Again, this is very highlevel and I am happy to go into the details.

In the meantime, I’ve got an OpenSource implementation for this here:

https://github.com/vaultie/lds-merkle-proof-2019

Deciding the proofValue will give you the transactionHas you are looking for.

The verificationMethod serves as a pointer to which DID was used to create this proofValue for the credential, in other words to verify **who** exactly issued the credential, not only to prove that it was issued.

And this is where you need ECDSA for example. Because you can prove that this transaction was indeed signed by the said wallet which belongs to some person or organization.

Reply:
if the transactionHash is in the proofValue and is checked against the verification method, that makes sense.

Reply:
This is up to the verifier to check that and I guess MerkleRoot2019 is specifically designed to do so.

Please, if you have any other questions, I’ll try to answer as soon as I can.